### PR TITLE
fix(gents): ignore @constructor annotations if inside goog.defineClass

### PR DESCRIPTION
**Pull request by bowenni**
Previously we get a nested class:
```javascript
class A {
  constructor: any = class {  };
}
```

### CODE REVIEW

File: src/main/java/com/google/javascript/gents/TypeConversionPass.java [hunk 1]
```diff
@@ -418,6 +419,17 @@ void convertDefineClassToClass(Node n) {
...
+  boolean inGoogDefineClass(Node n) {
+    if (n == null) {
+      return false;
+    }
+
+    if (n.isCall() && "goog.defineClass".equals(n.getFirstChild().getQualifiedName())) {
+      return true;
+    }
+
+    return inGoogDefineClass(n.getParent());
```
Comment (rkirov) [inline src/main/java/com/google/javascript/gents/TypeConversionPass.java]: I would skip traversing up here. What if you have something deeply nested what kinda made sense (I am making up stuff, but you never know):

```
goog.defineClass(null, {
  foo: function() {
    callMyApiThatExpectsAnonymousClass(/** @constructor */ function () {});
});
```

I think this should be a bit more restrictive and it should also check that the prop name is 'constructor'.

File: src/test/java/com/google/javascript/gents/singleTests/classes.js [hunk 2]
```diff
@@ -75,6 +75,17 @@ const F = goog.defineClass(null, {
...
 });
 
+/**
+ * goog.defineClass with @constructor annotation
+ */
+var GoogDefinedClassWithConstructorAnnotation = goog.defineClass(null, {
+  /**
+   * @constructor
+   */
+  constructor: function() {
```
Comment (rkirov) [inline src/test/java/com/google/javascript/gents/singleTests/classes.js]: add a test with params, so that the ctor function is kept.

Comment (bowenni) [inline src/test/java/com/google/javascript/gents/singleTests/classes.js]: Done.

File: src/main/java/com/google/javascript/gents/TypeConversionPass.java (hunk 1 above)
Comment (bowenni) [inline src/main/java/com/google/javascript/gents/TypeConversionPass.java]: Done.

**Pull request merged**